### PR TITLE
Fix fetch comments

### PR DESCRIPTION
**Pull request by apelisse**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/1676)

### CODE REVIEW

File: mungegithub/github/github.go [hunk 1]
```diff
@@ -1589,23 +1606,40 @@ func (obj *MungeObject) ListComments(withListOpts ...WithListOpt) ([]*github.Iss
...
+				// Cached last page was actually truthful -- expected error.
+				break
+			}
 			return nil, err
 		}
+		if tryNextPageAnyway {
+			if len(comments) == 0 {
+				break
+			}
+			glog.Infof("For %v: supposedly there weren't more comments, but we asked anyway and found %v more.", issueNum, len(comments))
```
Comment (eparis) [inline mungegithub/github/github.go]: can you distinguish these glog's so we know if it it only 1 or both of the APIs that are broken?

Comment (apelisse) [inline mungegithub/github/github.go]: True

**Comment by eparis** [comment]
@lavalamp it lgtm but you also helped run down this bug in the Events API.

**Comment by apelisse** [comment]
Fixed the nit,

**Comment by lavalamp** [comment]
/lgtm

but maybe you can scan the logs and make sure this is actually necessary.

**Pull request merged**